modules/shortcodes/dailymotion.php

```diff
@@ -327,3 +327,61 @@ function jetpack_dailymotion_embed_reversal( $content ) {
 }
 
 add_filter( 'pre_kses', 'jetpack_dailymotion_embed_reversal' );
+
+/**
+ * DailyMotion Widget Shortcode
+ *
+ * Example:
+ * [dailymotion-widget id=XYZ]
+ *
+ * @param array $atts
+ * @return string html
+ */
+function jetpack_dailymotion_widget_shortcode( $atts ) {
+	if ( isset( $atts['id'] ) ) {
+		$id = $atts['id'];
+	} else {
+		return '<!--Dailymotion error: bad or missing ID-->';
+	}
+
+	$output="<div class='dailymotion-widget' data-placement='" . $id . "'></div>";
+	$output .= '<script>(function(w,d,s,u,n,e,c){w.PXLObject = n; w[n] = w[n] || function(){(w[n].q = w[n].q || []).push(arguments);};w[n].l = 1 * new Date();e = d.createElement(s); e.async = 1; e.src = u;c = d.getElementsByTagName(s)[0]; c.parentNode.insertBefore(e,c);})(window, document, "script", "//api.dmcdn.net/pxl/client.js", "pxl");</script>';
+
+	return $output;
+}
+
+add_shortcode( 'dailymotion-widget', 'jetpack_dailymotion_widget_shortcode' );
+
+/**
+ * Embed Reversal for Dailymotion widget
```

Comment:
see https://github.com/Automattic/jetpack/pull/4550/commits/e925b0d9bdae84589c311c0f930cf767ed0646f4 
following your advice, I added @return and @param annotations on all the phpdocs where they were missing in this file